apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableVersion.cs

```diff
@@ -0,0 +1,199 @@
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Note: when using ReadModifyWriteRow, modified columns automatically use a server version, which
+        /// is based on the current timestamp micros. For those columns, other reads and writes should
+        /// use <see cref="BigtableVersion"/> values constructed from DateTime values, as opposed to using
+        /// a custom versioning scheme with 64-bit values.
+        /// </para>
+        /// </remarks>
+        /// <param name="timestamp">
+        /// The timestamp whose microseconds should be used as the version value. It must be specified in UTC.
+        /// </param>
+        public BigtableVersion(DateTime timestamp)
+        {
+            GaxPreconditions.CheckArgument(
+                timestamp.Kind == DateTimeKind.Utc,
+                nameof(timestamp),
+                $"The {nameof(BigtableVersion)} timestamp must be specified in UTC.");
+
+            Value = timestamp.Ticks / TicksPerMicro;
+        }
+
+        /// <summary>
+        /// Gets the version value. Greater version values indicate newer cell values.
+        /// </summary>
+        public long Value { get; }
+
+        /// <summary>
+        /// Gets the DateTime equivalent to the version assuming the value is a timestamp micros value.
+        /// </summary>
+        /// <returns>The DateTime representing the version timestamp.</returns>
+        public DateTime ToDateTime() => new DateTime(Value * TicksPerMicro, DateTimeKind.Utc);
+
+        /// <inheritdoc />
+        public int CompareTo(object obj)
+        {
+            if (obj is BigtableVersion other)
+            {
+                return CompareTo(other);
+            }
+            throw new ArgumentException($"The specified object cannot be compared with {nameof(BigtableVersion)}", nameof(obj));
+        }
+
+        /// <inheritdoc />
+        public int CompareTo(BigtableVersion other) => Value.CompareTo(other.Value);
+
+        /// <summary>
+        /// Compares two nullable <see cref="BigtableVersion"/> values.
+        /// </summary>
+        /// <param name="x">Left value to compare</param>
+        /// <param name="y">Right value to compare</param>
+        /// <returns>true if <paramref name="x"/> is less than <paramref name="y"/>; otherwise false.</returns>
+        public static int Compare(BigtableVersion? x, BigtableVersion? y)
+        {
+            if (x == null)
+            {
+                return y == null ? 0 : -1;
+            }
+            else if (y == null)
+            {
+                return 1;
+            }
+
+            return x.Value.CompareTo(y.Value);
+        }
+
+        /// <inheritdoc />
+        public bool Equals(BigtableVersion other) => CompareTo(other) == 0;
+
+        /// <inheritdoc />
+        public override bool Equals(object obj) => obj is BigtableVersion other && Equals(other);
```

Comment:
Just `Equals(other as BigtableVersion)` given the null result works? (No particular need either way.)

Reply:
`BigtableVersion` is a struct, so it isn't nullable.

Reply:
Doh - sorry about that :)